marshmallow_recipe/generics.py

```diff
@@ -0,0 +1,119 @@
+import dataclasses
+import types
+import typing
+from typing import Annotated, Any, Generic, TypeAlias, TypeVar, Union, get_args, get_origin
+
+_GenericAlias: TypeAlias = typing._GenericAlias  # type: ignore
+
+TypeLike: TypeAlias = type | TypeVar | types.UnionType | types.GenericAlias | _GenericAlias
+FieldsTypeMap: TypeAlias = dict[str, TypeLike]
+TypeVarMap: TypeAlias = dict[TypeVar, TypeLike]
+FieldsClassMap: TypeAlias = dict[str, TypeLike]
+ClassTypeVarMap: TypeAlias = dict[TypeLike, TypeVarMap]
+FieldsTypeVarMap: TypeAlias = dict[str, TypeVarMap]
+
+
+def get_fields_type_map(t: TypeLike) -> FieldsTypeMap:
+    origin = get_origin(t) or t
+    if not dataclasses.is_dataclass(origin):
+        return {}
```

Comment:
Should we even do these checks? We already did them
If we should, then maybe we should throw an exception?

Reply:
this is what I can suggest as improvement https://github.com/anna-money/marshmallow-recipe/pull/163/commits/4bbccfd80d0e34ec4b43905a0e389ae94a34f308

Reply:
Nice